File: codeforces/1600~1699/1687/a.cc
#include <bits/stdc++.h>

using namespace std;

using i64 = long long;

void solve() {
  int n = 0;
  i64 k = 0;
  cin >> n >> k;

  vector<i64> a(n);
  for (int i = 0; i < n; ++i) cin >> a[i];

  if (n == 1) {
    i64 ans = a[0] + k - 1;
    cout << ans << endl;
    return;
  }

  if (k < n) {
    i64 sum = 0;
    i64 ans = 0;
    for (int i = 0; i < n; ++i) {
      sum += a[i];
      if (i >= k) {
        sum -= a[i-k];
      }
      if (sum >= k-1) {
        ans = max(ans, sum);
      }
    }
    ans += k * (k-1) / 2;
    cout << ans << endl;
  } else if (k == n) {
    i64 ans = accumulate(a.begin(), a.end(), 0LL);
    ans += k * (k-1) / 2;
    cout << ans << endl;
  } else {
    i64 ans = accumulate(a.begin(), a.end(), 0LL);
    ans += (k-1 + k-n) * n / 2;

    cout << ans << endl;
  }
}

int main(void) {
  ios::sync_with_stdio(false);
  cin.tie(0);

  int T = 0;
  cin >> T;
  for (int t = 0; t < T; ++t) {
    solve();
  }
  return 0;
}

File: codeforces/1600~1699/1687/a.rs
use std::io;
use std::str::FromStr;
use std::fmt::Debug;
use std::cmp::max;

fn read_int<T: FromStr>() -> T where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    io::stdin().read_line(&mut s).unwrap();
    s.trim().parse::<T>().expect("invalid input")
}

fn read_vec<T: FromStr>() -> Vec<T> where <T as FromStr>::Err: Debug {
    let mut s = String::new();
    io::stdin().read_line(&mut s).unwrap();
    s.split(" ").map(|x| x.trim().parse().expect("invalid input")).collect()
}

fn solve() {
    let inputs = read_vec::<i64>();
    let n = inputs[0] as usize;
    let k = inputs[1];
    let a = read_vec::<i64>();

    if k <= n as i64 {
        let mut p: Vec<i64> = vec![0];
        for i in 0..n {
            p.push(p.last().unwrap() + a[i]);
        }
        let mut ans: i64 = 0;
        for i in 0..=n-(k as usize) {
            ans = max(ans, p[i+(k as usize)] - p[i]);
        }
        ans += k * (k-1) / 2;
        println!("{}", ans);
    } else {
        let mut ans: i64 = a.iter().sum();
        ans += (k-1 + k-(n as i64)) * (n as i64) / 2; 
        println!("{}", ans);
    }
}

fn main() {
    let n = read_int::<i32>();
    for _t in 0..n {
        solve();
    }
}
